apps/index.android.js

```diff
@@ -1,52 +1,133 @@
 var {
   AppRegistry,
   StyleSheet,
   Text,
   View,
-} = React;
+  TouchableNativeFeedback,
+  PullToRefreshViewAndroid,
+  } = React;
 
 var apps = React.createClass({
-  render: function() {
+  render: function () {
+    var button = <View />;
+    var showButton = this.state.loaded && !this.state.isAlive && !this.state.wakingPi && !this.state.stateError;
+    if (showButton) {
+      button =
+        <View>
+          <TouchableNativeFeedback onPress={this.buttonClicked} background={TouchableNativeFeedback.Ripple()}>
+            <View style={styles.button}>
+              <Text style={styles.buttonText}>Start con-nas</Text>
+            </View>
+          </TouchableNativeFeedback>
+        </View>
+    }
+
     return (
-      <View style={styles.container}>
-        <Text style={styles.welcome}>
-          Welcome to React Native!
-        </Text>
-        <Text style={styles.instructions}>
-          To get started, edit index.android.js
-        </Text>
-        <Text style={styles.instructions}>
-          Shake or press menu button for dev menu
-        </Text>
-      </View>
+      <PullToRefreshViewAndroid
+        style={styles.pullToRefreshViewAndroid}
+        refreshing={!this.state.loaded}
+        onRefresh={this.onRefresh}
+      >
+        <View style={styles.container}>
+          <Text>{this.state.loaded ? '' : 'loading...'}</Text>
+          <Text>{this.state.stateError ? 'error getting con-nas state' : ''}</Text>
+          <Text>{this.state.isAlive ? 'con-nas is up' : ''}</Text>
+          <Text>{this.state.feedback}</Text>
+          {button}
+        </View>
+      </PullToRefreshViewAndroid>
     );
-  }
+  },
+  componentDidMount: function () {
+    this.getConNasState();
+  },
+  getInitialState: function () {
+    return {
+      wakingPi: false,
+      loaded: false,
+      feedback: '',
+      isAlive: null,
+    };
+  },
+  buttonClicked: function () {
+    this.setState({feedback: '...'});
+    this.wakeConNas();
+  },
+  getConNasState: function () {
+    fetch(config.baseUrl + '/con-nas', {method: 'GET'})
+      .then((response) => response.json())
+      .then((responseData) => {
+        this.setState({loaded: true, isAlive: responseData.isAlive});
+      })
+      .catch((error) => {
+        this.setState({loaded: true, stateError: true});
+      })
+      .done();
+  },
+  // This is a dirty mock function to easily test random states from the server
+  getConNasStateMock: function () {
+    var rnd = Math.floor(Math.random() * (2));
+    this.setState({loaded: true, isAlive: rnd > 0});
+  },
+  wakeConNas: function () {
+    var _this = this;
+    this.setState({wakingPi: true});
+    fetch(config.baseUrl + '/con-nas', {
+      method: 'POST',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({})
+    })
+      .then((response) => response.json())
+      .then((responseData) => {
```

Comment:
Time for you to learn a nice trick here.
Ever used the ```Function.prototype.bind(env)``` method?

Every time you need to do a ```self = this``` (yeah ```self```is cleaner than ```_this``` IMO) you can replace it by a call to bind.
Bind enables you to specify which environment to use while executing the function.
So you would do something like:
```
.then(this.callback.bind(this))
// ...
// somewhere else in the "class"
callback: (regular, arguments) {
  // do stuffs
}

It is so much cleaner, because you only ever use ```this```ans you can move some code out of a sub-scope while keeping the closure context.
```